app/models/king.rb

```diff
@@ -1,5 +1,48 @@
 class King < Piece
 
+  def valid_move?(new_x, new_y)
+    #checks if the move that the king is making is valid
+    if position_exist?(new_x, new_y) && pass_king_rules?(new_x, new_y)
+      return true
+    else
+      return false
+    end
+  end
+
+  def position_exist?(new_x, new_y)
+    #checks if position that king is moving to exists
+    if new_x < 0 || new_x > 7 || new_y < 0 || new_y > 7
+      return false
+    else
+      return true
+    end
+  end
+
+  def allowed_positions
```

Comment:
Because the King can move in any direct exactly 1 steps, if that step is on the board & empty, is there a way to refactor this to reflect that easier? something like x_cord +/- 1, y_cord +/- 1.  I'll think more how to implement if you think this is a better refactor approach.

Not sure if there's a need to go in depth on refactoring that now, but just thinking out loud.